tests/e2e/odh_manager_test.go

```diff
@@ -4,116 +4,74 @@ import (
-		require.NoErrorf(t, tc.validateCRD("trustyais.components.platform.opendatahub.io"),
-			"error in validating CRD : trustyais.components.platform.opendatahub.io")
-	})
+	// Initialize the test context.
+	tc, err := NewTestContext(t)
+	require.NoError(t, err, "Failed to initialize test context")
 
-	t.Run("Validate Kueue CRD", func(t *testing.T) {
-		t.Parallel()
-		require.NoErrorf(t, tc.validateCRD("kueues.components.platform.opendatahub.io"),
-			"error in validating CRD : kueues.components.platform.opendatahub.io")
-	})
+	// Create an instance of test context.
+	operatorTestCtx := OperatorTestCtx{
+		TestContext: tc,
+	}
 
-	t.Run("Validate TrainingOperator CRD", func(t *testing.T) {
-		t.Parallel()
-		require.NoErrorf(t, tc.validateCRD("trainingoperators.components.platform.opendatahub.io"),
-			"error in validating CRD : trainingoperators.components.platform.opendatahub.io")
-	})
+	// Define test cases.
+	testCases := []TestCase{
+		{name: "Validate RHOAI Operator pod", testFn: operatorTestCtx.testODHDeployment},
+		{name: "Validate CRDs owned by the operator", testFn: operatorTestCtx.ValidateOwnedCRDs},
+	}
 
-	t.Run("Validate FeastOperator CRD", func(t *testing.T) {
-		t.Parallel()
-		require.NoErrorf(t, tc.validateCRD("feastoperators.components.platform.opendatahub.io"),
-			"error in validating CRD : feastoperators.components.platform.opendatahub.io")
-	})
-
-	t.Run("Validate DataSciencePipelines CRD", func(t *testing.T) {
-		t.Parallel()
-		require.NoErrorf(t, tc.validateCRD("datasciencepipelines.components.platform.opendatahub.io"),
-			"error in validating CRD : datasciencepipelines.components.platform.opendatahub.io")
-	})
-
-	t.Run("Validate Workbenches CRD", func(t *testing.T) {
-		t.Parallel()
-		require.NoErrorf(t, tc.validateCRD("workbenches.components.platform.opendatahub.io"),
-			"error in validating CRD : workbenches.components.platform.opendatahub.io")
-	})
-
-	t.Run("Validate Kserve CRD", func(t *testing.T) {
-		t.Parallel()
-		require.NoErrorf(t, tc.validateCRD("kserves.components.platform.opendatahub.io"),
-			"error in validating CRD : kserves.components.platform.opendatahub.io")
-	})
+	// Run the test suite.
+	RunTestCases(t, testCases)
+}
 
-	t.Run("Validate ModelMeshServing CRD", func(t *testing.T) {
-		t.Parallel()
-		require.NoErrorf(t, tc.validateCRD("modelmeshservings.components.platform.opendatahub.io"),
-			"error in validating CRD : modelmeshservings.components.platform.opendatahub.io")
-	})
+// testODHDeployment checks if the ODH deployment exists and is correctly configured.
+func (tc *OperatorTestCtx) testODHDeployment(t *testing.T) {
+	t.Helper()
 
-	t.Run("Validate ModelController CRD", func(t *testing.T) {
-		t.Parallel()
-		require.NoErrorf(t, tc.validateCRD("modelcontrollers.components.platform.opendatahub.io"),
-			"error in validating CRD : modelcontrollers.components.platform.opendatahub.io")
-	})
+	// Verify if the operator deployment is created
+	controllerDeployment := "rhods-operator"
```

Comment:
I had to manually set the operator controller deployment name to `rhods-operator` to ensure the corresponding E2E test checking the deployment readiness would pass. I'm currently working on a way to remove this dependency so that we don't have to hardcode or manually switch the deployment name between the `main` and `rhoai` branches.